include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -164,7 +164,17 @@ class ValueLoc
 namespace operators
 {
 template <typename T, typename IndexType, bool B>
-struct limits<::RAJA::reduce::detail::ValueLoc<T, IndexType, B>> : limits<T> {
+struct limits<::RAJA::reduce::detail::ValueLoc<T, IndexType, B>> {
+  RAJA_INLINE RAJA_HOST_DEVICE static constexpr
+  ::RAJA::reduce::detail::ValueLoc<T, IndexType, B> min()
+  {
+    return ::RAJA::reduce::detail::ValueLoc<T, IndexType, B>(limits<T>::min());
```

Comment:
I'm confused about a couple of things here.

1) These methods are creating `ValueLoc` objects initialized to some default state so they can be used in comparisons with other `ValueLoc` objects, correct?  If so, then should the `min` method initialize pass `limits<t>::max` and vice versa for the `max` method below?

2) Should we also be initializing the `loc` to some default? Not sure what that would be.

Reply:
This is a specialization of our equivalent to numeric_limits, so the min should really be the min and vice-versa.
1. Yes, but its the operator class that determines which value to use for its identity, see minimum/maximum.
2. Loc is initialized in the ValueLoc class itself.
I'm making this specialization more explicit because the old implementation used the base class implementation `limits<T>` and then relied on the "implicit" single value constructor of the ValueLoc class to promote that to the right type where needed. This seems potentially dangerous as it returned a T instead of an actual ValueLoc<T>.

Reply:
Got it. Thank you for the explanation.